src/resolve/resolved-dns-transaction.c

```diff
@@ -2301,7 +2301,7 @@ static int dns_transaction_add_dnssec_transaction(DnsTransaction *t, DnsResource
         return 1;
 }
 
-static int dns_transaction_request_dnssec_rr(DnsTransaction *t, DnsResourceKey *key) {
+static int dns_transaction_request_dnssec_rr_full(DnsTransaction *t, DnsResourceKey *key, DnsTransaction **ret) {
```

Comment:
hm, this function has multiple success paths, but only one initialize the return param.

our usual rule is that functions initialize all return params in all success paths...

Reply:
Already addressed in #31974 and it's follow-ups.

Reply:
i see now this already got addressed in a later commit. Ignore me then.